rllib/core/rl_trainer/trainer_runner.py

```diff
@@ -95,26 +90,38 @@ def __init__(
 
             # run the neural network building code on remote workers
             ray.get([w.build.remote() for w in self._workers])
+            # use only 1 max in flight request per worker since training workers have to
+            # be synchronously executed.
+            self._worker_manager = FaultTolerantActorManager(
+                self._workers,
+                max_remote_requests_in_flight_per_actor=1,
+            )
+            self._in_queue = deque(maxlen=20)
 
     @property
     def is_local(self) -> bool:
         return self._is_local
 
-    def update(self, batch: MultiAgentBatch) -> List[Mapping[str, Any]]:
+    def update(
+        self, batch: MultiAgentBatch, block: bool = True
+    ) -> List[Mapping[str, Any]]:
         """Do a gradient based update to the RLTrainer(s) maintained by this TrainerRunner.
 
         Args:
             batch: The data to use for the update.
+            block: Whether to block until the update is complete.
 
         Returns:
             A list of dictionaries of results from the updates from the RLTrainer(s)
         """
         if self.is_local:
             return [self._trainer.update(batch)]
         else:
-            return self._distributed_update(batch)
+            return self._distributed_update(batch, block=block)
```

Comment:
We should raise an error when block=False and local=True. It's invalid arg.

Reply:
done